main.go

```diff
@@ -47,14 +47,24 @@ func runMain() error {
 		return err
 	}
 
-	camera := lepton3.New(conf.SPISpeed)
-	err = camera.Open()
-	if err != nil {
-		return err
+	for {
+		camera := lepton3.New(conf.SPISpeed)
+		err = camera.Open()
+		if err != nil {
+			return err
+		}
+		defer camera.Close()
```

Comment:
Defers don't get run until the function returns. Given that main never returns these are just going to build up and not do anything.

Reply:
This is a common Go rookie error :)